### PR TITLE
update (core\store): sLFGDungeonStore update

### PR DESCRIPTION
**Pull request by acidmanifesto**
Update to proper struct

## Changes Proposed:
-  Updates the sLFGDungeonStore to propper struct
- Currently because of incorrect struct names, the info in the table does nothing, this PR corrects that.
- worth noting, dbcs that are loaded on AC must have a _dbc table, if one can't be provided then everything dealing with subsituting that dbc file and its info becomes a hardcoded hack fix mess.

## Issues Addressed:

- Closes 

## Tests Performed:

- builds
- runs

## How to Test the Changes:

1. well i suppose you can change a exsisting instance up a bit in the lfgdungeons_dbc
Not apart of the sql in this pr but i changed Pit of saraon to have the same level and flags as rage fire chasm and i ended up with this on my level 80 priest. not able to go in. Says level 80 though still on the client side.
image

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by acidmanifesto** [comment]
Cyrrent failure is with autobalance. minlevel ->MinLevel

File: data/sql/updates/pending_db_world/LfgDungeons.sql
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/LfgDungeons.sql:1]: can you fix the SQL so it is one column per line?

File: src/server/game/DataStores/DBCStores.cpp [hunk 1]
```diff
@@ -844,9 +844,15 @@ CharStartOutfitEntry const* GetCharStartOutfitEntry(uint8 race, uint8 class_, ui
...
 {
-    for (LFGDungeonEntry const* dungeon : sLFGDungeonStore)
-        if (dungeon->map == int32(mapId) && Difficulty(dungeon->difficulty) == difficulty)
+    for (uint32 i = 0; i < sLFGDungeonStore.GetNumRows(); ++i)
+    {
+        LFGDungeonEntry const* dungeon = sLFGDungeonStore.LookupEntry(i);
+        if (!dungeon)
+            continue;
+
+        if (dungeon->MapID == int32(mapId) && Difficulty(dungeon->Difficulty) == difficulty)
```
Comment (Kitzunu) [inline src/server/game/DataStores/DBCStores.cpp:853]: why do you need to convert it to int32? uin32 is just fine?

File: src/server/game/DungeonFinding/LFGMgr.h [hunk 2]
```diff
@@ -394,10 +394,10 @@ namespace lfg
...
         LFGDungeonData():  name("")
         { }
-        LFGDungeonData(LFGDungeonEntry const* dbc): id(dbc->ID), name(dbc->name[0]), map(dbc->map),
-            type(dbc->type), expansion(dbc->expansion), group(dbc->grouptype),
-            minlevel(dbc->minlevel), maxlevel(dbc->maxlevel), difficulty(Difficulty(dbc->difficulty)),
-            seasonal(dbc->flags & LFG_FLAG_SEASONAL), x(0.0f), y(0.0f), z(0.0f), o(0.0f)
+        LFGDungeonData(LFGDungeonEntry const* dbc) : id(dbc->ID), name(dbc->Name[0]), map(dbc->MapID),
+            type(dbc->TypeID), expansion(uint8(dbc->ExpansionLevel)), group(uint8(dbc->GroupID)),
+            minlevel(uint8(dbc->MinLevel)), maxlevel(uint8(dbc->MaxLevel)), difficulty(Difficulty(dbc->Difficulty)),
+            seasonal((dbc->Flags& LFG_FLAG_SEASONAL) != 0), x(0.0f), y(0.0f), z(0.0f), o(0.0f)
```
Comment (Kitzunu) [inline src/server/game/DungeonFinding/LFGMgr.h:400]:
```suggestion
            seasonal((dbc->Flags & LFG_FLAG_SEASONAL) != 0), x(0.0f), y(0.0f), z(0.0f), o(0.0f)
```

File: src/server/game/DataStores/DBCStores.cpp (hunk 1 above)
Comment (acidmanifesto) [reply src/server/game/DataStores/DBCStores.cpp:853]: Do we have a case where the interger will be negative?

File: data/sql/updates/pending_db_world/LfgDungeons.sql
Comment (acidmanifesto) [reply data/sql/updates/pending_db_world/LfgDungeons.sql:1]: Formatting fixed.

File: src/server/game/DataStores/DBCStores.cpp (hunk 1 above)
Comment (acidmanifesto) [reply src/server/game/DataStores/DBCStores.cpp:853]: unless you are hinting i should change
```
LFGDungeonEntry const* GetLFGDungeon(uint32 mapId, Difficulty difficulty)
```
to
```
LFGDungeonEntry const* GetLFGDungeon(int32 mapId, Difficulty difficulty)
``` 
and respected spots? but that if was already like that originally.

**Comment by acidmanifesto** [comment]
At the moment the only failure in the workflow is with autobalance in the custom module workflow.
image
Where minlevel needs to be changed to MinLevel per dbc column name. Other then that, this works fully as intended.
However on the 29th i will be no longer maintaining the pr and will be closed then. Anyone is free to take it over if that comes by. The purpose of this PR is to utlitze the actual propper column names of the LFGDungeonEntry DBC, for some reason we oddly made up our own names instead of using the propper ones.

**Comment by Kitzunu** [comment]
Have you tested it?

**Comment by acidmanifesto** [comment]
> Have you tested it?

Was tested on my local. I provided a screen shot in the how to test where i adjusted a lvl 77 - 80 i instance to match a much lower range instance on my lvl 80 priest.

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**